test/mocks/buffer/mocks.h

```diff
@@ -38,9 +39,9 @@ template <class BaseClass> class MockBufferBase : public BaseClass {
   }
 
   // A convenience function to invoke on write() which fails the write with EAGAIN.
-  int failWrite(int) {
+  std::tuple<int, int> failWrite(int) {
     errno = EAGAIN;
-    return -1;
+    return std::make_tuple(-1, errno);
```

Comment:
Ideally we can skip setting errno now, because no caller should be checking it.  If they are, that's a bug we can fix.

Reply:
Addressed in 36d2f52.